### PR TITLE
WebGLRenderer: Removed WebGLSpriteRenderer

### PR DESCRIPTION
**Pull request by Mugen87**
This is a first attempt to remove `WebGLSpriteRenderer` and integrate sprite rendering in `WebGLRenderer`.  Right now, `WebGLSpriteRenderer` does not support `ArrayCamera` and many other performance features (like uniform caching) of `WebGLRenderer`. Besides, the sprite material does not use any shader chunks from `three.js` shader lib. The PR should solve all this problems.

Should also fix #13073

### CODE REVIEW

**Comment by Mugen87** [comment]
Not sure if the shader programs are okay like that. We might want to remove/add certain chunks...

**Comment by WestLangley** [comment]
>Not sure if the shader programs are okay like that. We might want to remove/add certain chunks...

I think we can worry about that later.

This is looking pretty good to me.

File: src/renderers/WebGLRenderer.js [hunk 1]
```diff
@@ -1130,13 +1130,15 @@ function WebGLRenderer( parameters ) {
 
 		var opaqueObjects = currentRenderList.opaque;
 		var transparentObjects = currentRenderList.transparent;
+		var spritesObjects = currentRenderList.sprites;
```
Comment (WestLangley) [inline src/renderers/WebGLRenderer.js]: rename to `spriteObjects`?

**Comment by WestLangley** [comment]
I would not change it in this PR, but sprites are still rendered last -- even though they may, or may not, be transparent.

Is there a downside to rendering sprites along with the other transparent and opaque objects? Or would the use of sprite atlases make that approach too inefficient?

Maybe a solution is to continue to render sprites as a group, but have `opaqueSprites` and `transparentSprites` as separate arrays.

**Comment by WestLangley** [comment]
>I think sprite objects tend to share a common or at least similar rendering state. In order to achieve less WebGL state changes and more uniform cache hits, it might easier to achieve better performance if we manage sprites in a separate rendering list.

Well-stated.

However, I am still inclined to think we may need to render `opaqueObjects` first, then `opaqueSprites`, then `transparentObjects`, and finally `transparentSprites`.

If so, it can wait for a later PR.

**Comment by mrdoob** [comment]
Great stuff!

**Comment by mrdoob** [comment]
Thanks!

**Comment by Mugen87** [comment]
Yay! 🎉 

@WestLangley Thank you for triggering this change 😊

**Comment by sunag** [comment]
Very good. It seems an great opportunity for me update `sprites_nodes`

**Pull request merged**